deploy/docker-compose/README.md

```diff
@@ -0,0 +1,98 @@
+# Docker Compose Deployment
+
+## Quick Start
+
+This is a simple example that shows you how to connect `nm-vllm` with production monitoring via Prometheus and Grafana via Docker Compose.
+
+## Install
+
+Make sure you have Docker and Docker Compose installed.
+- [`docker`](https://docs.docker.com/engine/install/)
+- [`docker compose`](https://docs.docker.com/compose/install/linux/#install-using-the-repository)
+
+> Note: `nm-vllm` requires a GPU. Make sure you have installed the [Nvidia Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html).
+
+## Deploy
+
+### Launch
+
+The following spins up an `nm-vllm` instance, a Prometheus server, and a Grafana server:
+
+```bash
+docker compose up
+```
+
+### Simulate Client Requests
+
+Install requirements for client:
+```bash
+pip install -r client/requirements.txt
+```
+
+Download some sample data:
+```bash
+wget https://huggingface.co/datasets/anon8231489123/ShareGPT_Vicuna_unfiltered/resolve/main/ShareGPT_V3_unfiltered_cleaned_split.json
+```
+
+Launch clients:
+
+```bash
+python3 client/client.py --dataset-path ShareGPT_V3_unfiltered_cleaned_split.json --request-rate 1.0
```

Comment:
I'd suggest just `python`, but if you feel strongly about `python3`, we should probably update the `pip` above to be `pip3`. If we encourage them to create a `venv` then the `venv` will make appropriate aliases for `python` and `pip`, so the docs could just refer to `python` and `pip` consistently.